pkg/cmd/repo/clone/clone.go

```diff
@@ -87,6 +87,17 @@ func cloneRun(opts *CloneOptions) error {
 
 	var repo ghrepo.Interface
 	var protocol string
+	wikipref := false
+
+	// Check if repo opts has wiki in it and convert it into normal repo opts and get repo data
+	if strings.Contains(opts.Repository, ".wiki") {
```

Comment:
This check works, but will detect `.wiki` appearing anywhere in the repository argument, not just the end, which you would want to avoid, as we only ever want to match the `.wiki` suffix.

Since `opts.Repository` at this point can be a URL too, it's best to not try to match its name until after it's parsed. So, I would suggest moving this block to after the `if repositoryIsURL ... else` blocks, and having it be something like:

```go
wantsWiki := false
if strings.HasSuffix(repo.RepoName(), ".wiki") {
  repo = ghrepo.NewWithHost(repo.RepoOwner(), strings.TrimSuffix(repo.RepoName(), ".wiki"), repo.RepoHost())
  wantsWiki = true
}
```